src/staff.cpp

```diff
@@ -716,4 +718,22 @@ int Staff::GenerateMIDI(FunctorParams *functorParams)
     return FUNCTOR_CONTINUE;
 }
 
+int Staff::GenerateMIDIEnd(FunctorParams *functorParams)
+{
+    GenerateMIDIParams *params = vrv_params_cast<GenerateMIDIParams *>(functorParams);
+    assert(params);
+
+    // stop all previously held notes
+    for (auto &held : params->m_heldNotes) {
+        if (held.m_pitch > 0) {
+            params->m_midiFile->addNoteOff(params->m_midiTrack, held.m_stopTime * params->m_midiFile->getTPQ(),
+                params->m_midiChannel, held.m_pitch);
+        }
+    }
+
+    params->m_heldNotes.clear();
+
+    return FUNCTOR_CONTINUE;
+}
```

Comment:
This should be moved the the Layer. That is, move it to a `Layer::GenerateMIDIEnd(FunctorParams *functorParams)` with probably exactly the same code. I don't think we have seen encoding with multiple layers, but if we have some then the code should be triggered at the end of each layer and not of each staff (only).